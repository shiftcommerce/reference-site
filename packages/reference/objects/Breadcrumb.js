// Libraries
import { Component } from 'react'
import Link from 'next/link'

class Breadcrumb extends Component {
  constructor (props) {
    super(props)
    this.renderBreadcrumbs = this.renderBreadcrumbs.bind(this)
  }

  renderBreadcrumbs (trail) {
    return (
      trail && trail.map((crumb, idx) =>
        <li key={idx} className='o-breadcrumb__item'>
          <Link href={crumb.canonical_path} >
            <a className='o-breadcrumb__link'>{ crumb.title }</a>
          </Link>
        </li>
      )
    )
  }

  render () {
    let {
      trail,
      ...otherProps
    } = this.props

    return (
      <nav>
        <ol className={'o-breadcrumb'} {...otherProps}>
          <li className='o-breadcrumb__item'>
            <Link href='/home/index' as='/'>
              <a className='o-breadcrumb__link'>Home</a>
            </Link>
          </li>

          { this.renderBreadcrumbs(trail) }
        </ol>
      </nav>
    )
  }
}

export default Breadcrumb
