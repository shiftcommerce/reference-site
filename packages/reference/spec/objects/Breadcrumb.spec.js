// Object
import Breadcrumb from '../../objects/Breadcrumb'

// Fixtures
import breadcrumbMenuTrail from '../fixtures/breadcrumbs.fixture'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <Breadcrumb trail={breadcrumbMenuTrail} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})
