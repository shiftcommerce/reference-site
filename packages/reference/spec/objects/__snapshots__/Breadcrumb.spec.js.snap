// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Breadcrumb
  trail={
    Array [
      Object {
        "canonical_path": "/womens",
        "id": 1,
        "title": "Womens",
      },
      Object {
        "canonical_path": "/products/textured-long-t-shirt",
        "id": 2,
        "title": "Textured Long T-Shirt",
      },
    ]
  }
>
  <nav>
    <ol
      className="o-breadcrumb"
    >
      <li
        className="o-breadcrumb__item"
      >
        <Link
          href="/"
        >
          <a
            className="o-breadcrumb__link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </Link>
      </li>
      <li
        className="o-breadcrumb__item"
      >
        <Link
          href="/womens"
        >
          <a
            className="o-breadcrumb__link"
            href="/womens"
            onClick={[Function]}
          >
            Womens
          </a>
        </Link>
      </li>
      <li
        className="o-breadcrumb__item"
      >
        <Link
          href="/products/textured-long-t-shirt"
        >
          <a
            className="o-breadcrumb__link"
            href="/products/textured-long-t-shirt"
            onClick={[Function]}
          >
            Textured Long T-Shirt
          </a>
        </Link>
      </li>
    </ol>
  </nav>
</Breadcrumb>
`;
